lib/buffer.js

```diff
@@ -304,6 +306,20 @@ function slowToString(encoding, start, end) {
   if (end > this.length) end = this.length;
   if (end <= start) return '';
 
+  switch (encoding) {
+    case 'ucs2':
+    case 'ucs-2':
+    case 'utf16le':
+    case 'utf-16le':
+      if (((end - start) / 2) > kStringMaxLength)
+        throw new RangeError('invalid String length');
```

Comment:
What should I do if it failed after `MaybeLocal`? I wanted to throw an exception but bnoordhuis only wants exceptions  at the edges of the JS/C++ bridge? (Does we catch the context-free exception dcarney wants to implement?) So should I just check for an empty handle in StringSlice and then throw an exception? I tested for checking for `undefined` in `Buffer.prototype.toString` but that resulted in 200 ms more in the benchmark although it only adds `movq, cmpq, jnz`. Maybe returning undefined is enough :)